Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.h

```diff
@@ -124,7 +121,16 @@ class CDMInstanceOpenCDM final : public CDMInstance {
 private:
     media::OpenCdm m_openCDM;
     const char* m_mimeType;
-    std::map<std::string, Session> m_sessionsMap;
+
+    // Protects against concurrent access to m_sessionsMap. In addition to the main thread
+    // the GStreamer decryptor elements running in the streaming threads have a need to
+    // lookup values in this map.
+    mutable Lock m_sessionMapMutex;
+    HashMap<String, RefPtr<CDMInstanceOpenCDM::Session>> m_sessionsMap;
+    bool addSession(const String& sessionId, Session* session);
+    bool removeSession(const String& sessionId);
+    RefPtr<Session> lookupSession(const String& sessionId) const;
```

Comment:
I think there is not explicit rule for this, but it is nicer if you keep the methods at the top, just after the `private:` and the attributes at the bottom.